bindgen/codegen/mod.rs

```diff
@@ -2987,6 +2992,48 @@ impl CompInfo {
             }
         }
     }
+
+    /// Use a fallback comment from a type alias to this type if necessary
+    ///
+    /// The documentation for a type could get lost in the following circumstances:
+    ///
+    /// - We have a type and a type alias with the same canonical path
+    /// - The Documentation is only associated with the type alias
+    ///
+    /// In this case bindgen will not generate the type alias and the documentation would be lost.
+    /// To avoid this, we check here if there is any type alias to this type, which has
+    /// the same canonical path and return the comment as a fallback, if our type does
+    /// not have documentation.
+    fn get_typedef_fallback_comment(
+        ctx: &BindgenContext,
+        type_id: &crate::ir::context::TypeId,
+    ) -> Option<String> {
+        if !ctx.options().generate_comments {
+            return None;
+        }
+        let type_alias_comment = ctx
+            .items()
+            .filter(|(_id, alias)| {
```

Comment:
Iterating over all items seems not really fine, performance wise. Can we avoid it somehow?

Reply:
Hmm - I think since we are using this fallback for the case where typedef and struct declaration are at the same time, a potential type alias should always be at `item_id + 1`. I don't think we currently have a way to access the "next" item or do arithmetic on item ids, but this would avoid iterating through all items.

Reply:
Maybe we can do it the other way around? During the clang phase, go peek at the typedef's comment? Haven't tried it tho.